File: app/src/main/java/com/github/premnirmal/ticker/portfolio/EditPositionActivity.java
package com.github.premnirmal.ticker.portfolio;

import android.content.DialogInterface;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import com.github.premnirmal.ticker.network.Stock;
import com.github.premnirmal.tickerwidget.R;

/**
 * Created by premnirmal on 7/3/15.
 */
public class EditPositionActivity extends AddPositionActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final Stock stock = stocksProvider.getStock(ticker);
        if(stock != null) {
            final EditText sharesView = (EditText) findViewById(R.id.shares);
            final EditText priceView = (EditText) findViewById(R.id.price);
            final Button skipButton = (Button) findViewById(R.id.skipButton);
            skipButton.setText(R.string.remove);
            sharesView.setText("" + stock.PositionShares);
            priceView.setText("" + stock.PositionPrice);
        } else {
            showDialog(getString(R.string.no_such_stock_in_portfolio), new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    finish();
                }
            });
        }
    }

    @Override
    protected void skip() {
        stocksProvider.removePosition(ticker);
        super.skip();
    }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/portfolio/EditPositionActivity.kt
package com.github.premnirmal.ticker.portfolio

import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import com.github.premnirmal.ticker.AppPreferences
import com.github.premnirmal.tickerwidget.R
import kotlinx.android.synthetic.main.activity_positions.addRemoveText
import kotlinx.android.synthetic.main.activity_positions.averagePositionPrice
import kotlinx.android.synthetic.main.activity_positions.currentPosition
import kotlinx.android.synthetic.main.activity_positions.price
import kotlinx.android.synthetic.main.activity_positions.priceInputLayout
import kotlinx.android.synthetic.main.activity_positions.removeButton
import kotlinx.android.synthetic.main.activity_positions.shares
import kotlinx.android.synthetic.main.activity_positions.sharesInputLayout
import kotlinx.android.synthetic.main.activity_positions.skipButton
import kotlinx.android.synthetic.main.activity_positions.toolbar

/**
 * Created by premnirmal on 2/25/16.
 */
class EditPositionActivity : AddPositionActivity() {

  companion object {
    const val TICKER = "TICKER"
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    toolbar.setTitle(R.string.edit_position)
    val stock = stocksProvider.getStock(ticker)
    if (stock != null) {
      if (stock.isPosition) {
        currentPosition.text = stock.totalPosition.toString()
        averagePositionPrice.text = String.format("%.2f",stock.averagePositionPrice);
        removeButton.setOnClickListener{ onRemoveClicked() }
        removeButton.visibility = View.VISIBLE
        addRemoveText.setText(R.string.add_remove_position)
      }
    } else {
      showDialog(getString(R.string.no_such_stock_in_portfolio),
          DialogInterface.OnClickListener { _, _ -> finish() })
    }
  }

  protected fun onRemoveClicked() {
    val sharesView = shares
    val sharesText = sharesView.text.toString()
    if (!sharesText.isEmpty()) {
      var shares = 0f
      var success = true
      try {
        shares = sharesText.toFloat()
      } catch (e: NumberFormatException) {
        sharesInputLayout.error = getString(R.string.invalid_number)
        success = false
      }
      if (success) {
        priceInputLayout.error = null
        sharesInputLayout.error = null
        stocksProvider.decreasePosition(ticker, shares)
        finish()
      }
    }
  }
}